src/Forms/Schema/FormSchema.php

```diff
@@ -14,6 +16,47 @@
  */
 class FormSchema
 {
+    /**
+	 * Returns a representation of the provided {@link Form} as structured data,
+	 * based on the request data.
+	 *
+     * @param HTTPRequest $request
+	 * @param string $schemaID ID for this schema. Required.
+	 * @param Form $form Required for 'state' or 'schema' response
+	 * @param ValidationResult $result Required for 'error' response
+	 * @return array
+	 */
+	public function getMultipartSchema($request, $schemaID, Form $form = null, ValidationResult $result = null) {
```

Comment:
I think this particular method shouldn't concern itself with *how* the schema parts are requested - if we switch this to GraphQL sometime, it likely would be contained in the GraphQL query itself, rather than `$request`. Maybe we could accept an array of parts here instead, and have another helper method to extract that from a `HTTPRequest`?